test/unit-test/FreeRTOS_RA/FreeRTOS_RA_utest.c

```diff
@@ -0,0 +1,812 @@
+}
+
+/**
+ * @brief This function verify sending an Router Solicitation ICMPv6 message
+ *        With a pxDescriptor being NULL.
+ */
+void test_vNDSendRouterSolicitation_NullDesc( void )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer, xNetworkBuffer2;
+    NetworkEndPoint_t xEndPoint;
+    IPv6_Address_t xIPAddress;
+    ICMPPacket_IPv6_t * pxICMPPacket, xICMPPacket;
+    IPHeader_IPv6_t xIPHeader;
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxICMPPacket = &xICMPPacket;
+    pxNetworkBuffer->xDataLength = uxHeaderBytesRS - 1;
+    xNetworkBuffer2.xDataLength = uxHeaderBytesRS + 1;
+    pxNetworkBuffer->pxEndPoint = &xEndPoint;
+    pxNetworkBuffer->pucEthernetBuffer = &xICMPPacket;
+    xEndPoint.bits.bIPv6 = 1;
+    pxICMPPacket->xIPHeader = xIPHeader;
+
+    memset( &xEndPoint.ipv6_settings, 0, sizeof( IPV6Parameters_t ) );
+    memset( &pxICMPPacket->xIPHeader, 0, sizeof( IPHeader_IPv6_t ) );
+
+    FreeRTOS_FirstEndPoint_IgnoreAndReturn( &xEndPoint );
+    FreeRTOS_NextEndPoint_IgnoreAndReturn( NULL );
+
+    pxDuplicateNetworkBufferWithDescriptor_ExpectAnyArgsAndReturn( NULL );
+
+    vNDSendRouterSolicitation( pxNetworkBuffer, &xIPAddress );
+}
+
+/**
+ * @brief This function verify succeffully sending an
+ *        ICMPv6 message of the type: Router Solicitation
+ */
+void test_vNDSendRouterSolicitation_HappyPath( void )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer;
+    NetworkEndPoint_t xEndPoint;
+    IPv6_Address_t xIPAddress;
+    ICMPPacket_IPv6_t * pxICMPPacket, xICMPPacket;
+    IPHeader_IPv6_t xIPHeader;
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxICMPPacket = &xICMPPacket;
+    pxNetworkBuffer->xDataLength = uxHeaderBytesRS + 1;
+    pxNetworkBuffer->pxEndPoint = &xEndPoint;
+    pxNetworkBuffer->pucEthernetBuffer = &xICMPPacket;
+    xEndPoint.bits.bIPv6 = 1;
+    pxICMPPacket->xIPHeader = xIPHeader;
+
+    memset( &xEndPoint.ipv6_settings, 0, sizeof( IPV6Parameters_t ) );
+    memset( &pxICMPPacket->xIPHeader, 0, sizeof( IPHeader_IPv6_t ) );
+
+    FreeRTOS_FirstEndPoint_IgnoreAndReturn( &xEndPoint );
+    FreeRTOS_NextEndPoint_IgnoreAndReturn( NULL );
+
+    usGenerateProtocolChecksum_ExpectAnyArgsAndReturn( ipCORRECT_CRC );
+    vReturnEthernetFrame_ExpectAnyArgs();
+
+    vNDSendRouterSolicitation( pxNetworkBuffer, &xIPAddress );
+
+    TEST_ASSERT_EQUAL( pxICMPPacket->xIPHeader.ucVersionTrafficClass, raDEFAULT_VERSION_TRAFFIC_CLASS );
+    TEST_ASSERT_EQUAL( pxICMPPacket->xIPHeader.usPayloadLength, FreeRTOS_htons( sizeof( ICMPRouterSolicitation_IPv6_t ) ) );
+    TEST_ASSERT_EQUAL( pxICMPPacket->xIPHeader.ucNextHeader, ipPROTOCOL_ICMP_IPv6 );
+    TEST_ASSERT_EQUAL( pxICMPPacket->xIPHeader.ucHopLimit, raDEFAULT_HOP_LIMIT );
+}
+
+/* ===================================================
+ *           Test for vReceiveNA
+ * =================================================== */
+
+
+void test_vReceiveNA_NoEndPoint( void )
```

Comment:
Can we add a  small description of what the test does here as well similar to other tests.

Reply:
Yes, its in progress, i am adding the descriptions.